### ISSUE TITLE
preliminary code style and structure feedback

### ISSUE THREAD

**Issue by jbphet** [phetsims/masses-and-springs#70]
Since @Denz1994 is relatively new to PhET, we've decided to do some periodic intermediate code reviews so that he gets feedback on things like coding standards, architecture, and so forth, as the implementation progresses.  He and I just went through a review of a couple of files, here is a list of what we discussed (for followup):

File: `MassesAndSpringsModel.js`

- [x] comment: "TODO:: There has to be a better way to manage all of these properties as one entity without using propertySet()".  I'm not sure that there is, and I wouldn't sweat it.

- [x] strange comments, e.g. "c - coefficient of friction" and "a - gravitational acceleration (positive)"

- [x] Line 144, declaration of this.springs - clean up long lines

- [x] Lines 191 through 215 - comment the linkages

- [x] Function 'createArrow' created on line 206 - should be in "static block"

- [x] In function headers, put the viz annotation (e.g. @public) last, in a block with other annotations

- [x] comment code standards (saw some stuff in adjustDraggedMassPosition):
  - [x] blank line before comments
  - [x] only use capital letter if writing as a full sentence
  
- [x] in step function line 328 use _.values( this.masses ).forEach( function( mass ){} )

File: `Spring.js`

- [x] Header comment doesn't make much sense.  This is a model of a spring, right?  And it is limited to vertical motion, right?

- [x] line 91, declaration of positionProperty - what part of the spring does this correspond to?  The top?  It should say, and maybe be conveyed in the name.

- [x] line 101 - where do the values that define the range come from, and can they be made into constants for better readability/maintainability?

- [x] line 133 should be unnecessary since the value was set when the property was constructed.

- [x] Derived properties starting at line 139 - formatting needs some minor modifications

- [x] line 177 - use natural resting length like the other properties (as a parameter)

- [x] line 211 - the reset function has a parameter, but I don't see any calls to reset with a parameter.  What's up with this?

- [x] In the inherit block, we should follow the "Chris Convention" for header comments and put the viz annotation last (look through Function Builder for examples).

- [x] In my opinion, the oscillate: function should be step: in order to be more consistent with other model elements, or at least stepOscillation.

**Comment by Denz1994** [comment]
The management of the scene state has been completely refactored as per suggestion by @jbphet. It includes state management in the mass and spring objects respectively and relaying that back to the intro model. The previous attempt handled everything in the model. 

I was able to remove about 150 lines of code with this change and the code looks a lot more compartmentalized and readable.

There is still documentation to do a bit more clean up but the major features of state restoration have been implemented. Thanks for the suggestion @jbphet. After some more clean-up I will assign to @jbphet for review.
